requirements.txt

```diff
@@ -9,7 +9,7 @@ python-dateutil==2.8.2
 pyyaml==6.0.1
 httpx[http2]==0.24.1
 Brotli==1.1.0
-uvloop==0.17.0
+uvloop==0.17.0; python_version < '3.12'
```

Comment:
I made a `make clean` to build a fresh environment with my python `3.12.0rc3` runtime, but I do not understand why uvloop wheel is build .. AFAIK uvloop is only a requirement in `; python_version < '3.12'`

Here is the issue I have:

```
$ asdf shell python 3.12.0rc3
$ make clean
$ make test
...
Building wheels for collected packages: uvloop
  Building wheel for uvloop (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
```

May this is an quirk of semantic versioning where a releas candidate is *lower* than the release itself?

What is true in semantic versioning:

- `3.12 < 3.12.0-rc.3`
- `3.12.0-rc.3 < 3.12` or ..

When the latter one is true, I understand why python tries to build the uvloop wheel.

Reply:
> I do not understand why uvloop wheel is build

Because there is wheel for Python 3.12

Reply:
I don't have issue with uvloop:

```sh
$ python --version
Python 3.12.0rc3
$ make install
...
Ignoring uvloop: markers 'python_version < "3.12"' don't match your environment
...
```

Reply:
Hm strange / I looked at [PEP0440](https://peps.python.org/pep-0440/) .. and they say:

>  The following example covers many of the possible combinations:

```
1.dev0
1.0.dev456
1.0a1
1.0a2.dev456
1.0a12.dev456
1.0a12
1.0b1.dev456
1.0b2
1.0b2.post345.dev456
1.0b2.post345
1.0rc1.dev456
1.0rc1
1.0
...
```

I'm confused how it would be on a patch level 0 `3.12.0rc3` ..

Reply:
aiohttp seems to use `python_version < "3.12"`:  https://github.com/aio-libs/aiohttp/pull/7503/files

This puzzle me: why is it working differently. Is `python_version <= "3.11"` okay to say all Python 3.11 versions ?

Reply:
> This puzzle me: why is it working differently. 

I'm sooo sorry .. the issue sits in front of the PC .. I had local changes in my branch and was testing with the wrong conditions :facepalm: .. sorry for the noise.

Now with correctly set `python_version < "3.12"` uvloop is no longer installed .. and now I can reproduce the issue we see in the gh CI :+1:

---
OT: But the semantic order of PEP-440 confuses me --> they give this example of versions ordering:

```
1.0rc1
1.0
```

In this ordering `3.12.0rc3 < 3.12` should result to `True`.